### PR TITLE
Alternative changes to #115

### PR DESCRIPTION
**Pull request by gemmellr**
Alternative changes to #115

### CODE REVIEW

File: src/main/java/io/vertx/proton/impl/ProtonReceiverImpl.java [hunk 1]
```diff
@@ -231,6 +249,53 @@ void onDelivery() {
...
+
+  private void handleMaxMessageSizeExceeded(final UnsignedLong maxMessageSize, final Receiver receiver) {
+    try {
+      LOG.debug("delivery received exceeding max-message-size of " + maxMessageSize + " bytes");
+      if (maxMessageSizeExceededHandler != null) {
+        maxMessageSizeExceededHandler.handle(this);
+      }
+    } finally {
+      // Detach link if handler has not sent a detach frame already
+      if (!getReceiver().detached() && isOpen()) {
```
Comment (sophokles73) [inline src/main/java/io/vertx/proton/impl/ProtonReceiverImpl.java:291]: `!receiver.detached() && isOpen()` ?

**Review by sophokles73 — COMMENTED** [review]
Looks even better than my proposed change :+1: 
please feel free to pick (and change) my unit tests from #115 if they are helpful ...

File: src/main/java/io/vertx/proton/impl/ProtonReceiverImpl.java (hunk 1 above)
Comment (gemmellr) [reply src/main/java/io/vertx/proton/impl/ProtonReceiverImpl.java]: yep, meant to do that and missed it, updated now.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Comment by gemmellr** [comment]
I took your tests from #115 and updated them for the change to detaching, tweaked some of the values used and added some extra checks. I decided just to push those changes in now before I get to the other testing, as I believe a release is planned, and the entire difference sits behind a gating null check anyway and so could be easily avoided if needed.

Pushed to master in https://github.com/vert-x3/vertx-proton/commit/6955463580097a6d18cdc5df17b26c1464f56715 and backported to 3.9 in https://github.com/vert-x3/vertx-proton/commit/4090c6fabf91c5b5380a10220c61c06d240f3fb9.

**Pull request merged**